config/app_dynamics_agent.yml

```diff
@@ -17,6 +17,7 @@
 ---
 version: 4.1.+
 repository_root: ! '{default.repository.root}/app-dynamics'
-default_application_name:
-default_node_name: ! '$(expr "$VCAP_APPLICATION" : ''.*instance_index[": ]*\([[:digit:]]*\).*'')'
+default_application_name: 
+#default_node_name: ! '$(expr "$VCAP_APPLICATION" : ''.*instance_index[": ]*\([[:digit:]]*\).*'')'
+default_node_name: ! '$(expr "$VCAP_APPLICATION" : '.*"application_name[": ]*\\([a-z_0-9-]\\+\\).*')[$(expr "$VCAP_APPLICATION" : '.*"instance_index[": ]*\\([0-9]\\+\\).*')]-[provo]'
```

Comment:
I see two realistic alternatives:

```
! '$(expr "$VCAP_APPLICATION" : ''.*application_name[": ]*\([-_a-zA-Z0-9]*\).*'')[$(expr "$VCAP_APPLICATION" : ''.*instance_index[": ]*\([[:digit:]]*\).*'')]'
```

which gives you `java-main-application[0]` or 

```
! '$(expr "$VCAP_APPLICATION" : ''.*application_name[": ]*\([-_a-zA-Z0-9]*\).*''):$(expr "$VCAP_APPLICATION" : ''.*instance_index[": ]*\([[:digit:]]*\).*'')'
```

 which gives you `java-main-application:0`.  I prefer the second, but don't have a strong opinion.  Anyone else have an opinion?